pkg/engines/slurm/slurm.go

```diff
@@ -140,55 +141,52 @@ func GenerateOutput(ctx context.Context, tree *topology.Vertex, params map[strin
 
 func GenerateOutputParams(ctx context.Context, tree *topology.Vertex, params *Params) ([]byte, error) {
 	buf := &bytes.Buffer{}
-	path := params.TopoConfigPath
+	path, plugin := params.TopoConfigPath, params.Plugin
+
+	// set and validate plugin
+	switch plugin {
+	case "":
+		plugin = topology.TopologyTree
+	case topology.TopologyTree, topology.TopologyBlock:
+		// nop
```

Comment:
Also, if params provided plugin is Block but block subtree doesn't exist, unsupported message should be reported.

Reply:
addressed